src/selection/block_generator.rs

```diff
@@ -87,9 +87,10 @@ impl<T: Operations> Generator<T> {
             generator_bls_pubkey: *pubkey.bytes(),
             state_hash: [0; 32],
             hash: [0; 32],
+            cert: Certificate::default(),
         };
 
-        Ok(Block::new(blk_header, vec![]))
+        Ok(Block::new(blk_header, vec![]).expect("block should be valid"))
```

Comment:
Should we reprobavate the error instead of panicking?

Reply:
While testing is progress, I'd like to have fail-fast approach.